examples/hacker_news_assets/hacker_news_assets/repository.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+
+from dagster import (
+    repository,
+    with_resources,
+    AssetSelection,
+    build_schedule_from_partitioned_job,
+    define_asset_job,
+)
+
+from hacker_news_assets.assets import (
+    activity_analytics_assets,
+    ACTIVITY_ANALYTICS,
+    core_assets,
+    CORE,
+    recommender_assets,
+    RECOMMENDER,
+    dbt_assets,
+)
+from hacker_news_assets.resources import RESOURCES_LOCAL, RESOURCES_PROD, RESOURCES_STAGING
+from hacker_news_assets.sensors import make_slack_on_failure_sensor, make_hn_tables_updated_sensor
+from hacker_news_assets.partitions import hourly_partitions
+
+
+# TBD: is it a good practice to put all sensors and sensors live inside repository?
+# * pros:
+#       * no extra schedule and sensor layers at starter - sensors and schedules are mostly just wrappers around jobs or asset groups
+#       * users can choose to separate them out once the project grows
+# * cons:
+#       * it'd be a lot of code "in your face" as the the repository is the main entry of a dagster
+#         project.
```

Comment:
RFC

Reply:
I am in support of sensors and schedules living inside `repository.py`. I think they should be defined outside of the `@repository` function but within the file (like you did in this PR) to make the contents of the repository readable.

I think the inconvenience of having to do multiple imports if schedules/sensors existed in their own file outweighs the "code in your face"